dask/dataframe/core.py

```diff
@@ -1451,13 +1451,29 @@ def var(self, axis=None, skipna=True, ddof=1, split_every=False, dtype=None, out
                                     axis=axis, skipna=skipna, ddof=ddof)
             return handle_out(out, result)
         else:
-            num = self._get_numeric_data()
-            x = 1.0 * num.sum(skipna=skipna, split_every=split_every)
-            x2 = 1.0 * (num ** 2).sum(skipna=skipna, split_every=split_every)
-            n = num.count(split_every=split_every)
-            name = self._token_prefix + 'var'
-            result = map_partitions(methods.var_aggregate, x2, x, n,
-                                    token=name, meta=meta, ddof=ddof)
+            num = self.select_dtypes(include=['number', np.timedelta64, 'bool']) \
+                if is_dataframe_like(self) else self
+
+            values_dtype = num.values.dtype
```

Comment:
This is concerning me slightly. What happens when we have a heterogenous dataframe?

```python
In [6]: df = pd.DataFrame({"A": [1, 2, 3], "B": pd.to_timedelta([1, 2, 3])})

In [7]: ddf = dd.from_pandas(df, 2)

In [8]: df.var()
Out[8]:
A    1.0
B    1.0
dtype: float64

In [9]: ddf.var()
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-9-d17ffc2129b0> in <module>
----> 1 ddf.var()

~/sandbox/dask/dask/dataframe/core.py in var(self, axis, skipna, ddof, split_every, dtype, out)
   1467             cols = num._meta.columns if is_dataframe_like(num) else None
   1468
-> 1469             var_shape = num._meta_nonempty.values.var(axis=0).shape
   1470             array_var_name = (array_var._name,) + (0,) * len(var_shape)
   1471

~/Envs/dask-dev/lib/python3.7/site-packages/numpy/core/_methods.py in _var(a, axis, dtype, out, ddof, keepdims)
    119         x = um.multiply(x, um.conjugate(x), out=x).real
    120     else:
--> 121         x = um.multiply(x, x, out=x)
    122     ret = umr_sum(x, axis, dtype, out, keepdims)
    123

TypeError: unsupported operand type(s) for *: 'Timedelta' and 'Timedelta'
```

On master, it seems we incorrectly dropped timedeltas.

I think the more correct thing to do is to avoid the `.values` and apply things columnwise. That has a negative performance impact for very wide dataframe, as we'll have one `var` computation per column, rather than per dtype. I'll need to think a bit more about what to do here.

Reply:
@TomAugspurger omg, seems like I missed that test_reductions only includes tests for series and not for dataframe :'( Sorry for that. It is a good catch. I can do the same way as it's done in describe - if there is timedelta column in dataframe, apply columnwise, if not - per dtype. This way it's not going to affect performance for numeric dataframes. timedeltas first can be converted to float64 and then after computation is applied - back to timedelta.

Reply:
@TomAugspurger I've added per-column variance for dataframes with timedelta columns. For all-numerical everything remains the same.